src/section/section_map.rs

```diff
@@ -6,101 +6,88 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use super::{elders_info::EldersInfo, network_stats::NetworkStats};
+use super::{elders_info::EldersInfo, network_stats::NetworkStats, prefix_map::PrefixMap};
 use crate::{
+    consensus::Proven,
     id::P2pNode,
     location::DstLocation,
     xor_space::{Prefix, XorName},
 };
 use std::{
-    cmp::Ordering,
-    collections::{BTreeMap, BTreeSet, VecDeque},
+    collections::{BTreeSet, HashSet},
     iter,
 };
 
-// Number of recent keys we keep: i.e how many other section churns we can handle before a
-// message sent with a previous version of a section is no longer trusted.
-// With low churn rate, an ad hoc 20 should be big enough to avoid losing messages.
-const MAX_RECENT_KEYS: usize = 20;
-
 /// Container for storing information about sections in the network.
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub struct SectionMap {
     // Our section.
-    our: EldersInfo,
+    our: Proven<EldersInfo>,
     // Neighbour sections: maps section prefixes to their latest signed elders infos.
-    // Note that after a split, the section's latest section info could be the one from the
-    // pre-split parent section, so the value's prefix doesn't always match the key.
-    neighbours: BTreeMap<Prefix<XorName>, EldersInfo>,
-    // BLS public keys of known sections
-    keys: BTreeMap<Prefix<XorName>, bls::PublicKey>,
-    // Recent keys removed from `keys`. Contains at most `MAX_RECENT_KEYS` entries.
-    recent_keys: VecDeque<(Prefix<XorName>, bls::PublicKey)>,
-    // Indices of our section keys that are trusted by other sections.
-    knowledge: BTreeMap<Prefix<XorName>, u64>,
+    neighbours: PrefixMap<Proven<EldersInfo>>,
+    // BLS public keys of known sections excluding ours. The proof is for the whole `(prefix, key)`
```

Comment:
`The proof` refers to `Proven<(Prefix<XorName>, bls::PublicKey)>` , right?

Reply:
Yes. The comment was written before the introduction of the `Proven` type and it's a bit redundant now. I'll change it.